Stage 2

prathamponnana opened the pull request:
Included new image and new items. 
Also I've made the page responsive. I have used mobile-first approach here.
Please review my code sir.

prathamponnana committed:
Add files via upload

>New picture with caption and heading
>Added local storage feature

prathamponnana committed:
Update README.md

friendly-developer commented on app.js:
```diff
@@ -1,6 +1,21 @@
+let counter2 = 0;
+document.querySelector('.my-pic1').addEventListener('click', function(){ 
+    counter1++;
+    localStorage.setItem('counter1',counter1);
```
instead of setting two different local storage values, you can instead create an object and update it.
After every click, you can save the updated object to the local storage

prathamponnana replied:
Thank you sir. I have updated the code in the recent commit.

friendly-developer commented on app.js:
```diff
@@ -1,6 +1,21 @@
+document.querySelector('.my-pic2').addEventListener('click', function(){
+    counter2++;
+    localStorage.setItem('counter2',counter2);
+    document.getElementById('btn2').innerHTML= localStorage.getItem('counter2');
```
avoid using innerHTML, check for alternative methods of doing it, innerHTML is a security vulnerability

friendly-developer commented on index.html:
```diff
@@ -10,12 +10,27 @@
     <footer id="footer">
-        <div>Pratham Ponnana</div>
+        <hr>
+        <div>Created by <a href="https://www.linkedin.com/in/prathamponnana/">Pratham Ponnana</a></div>
```
good job adding a LinkedIn referral

friendly-developer commented on styles.css:
```diff
@@ -2,13 +2,40 @@
 
-.my-pic{
-    width:500px;
+.my-pic1,.my-pic2{
```
maybe you can create a parent class for both of these

prathamponnana committed:
Update index.html

prathamponnana committed:
Update styles.css

prathamponnana committed:
Update app.js

Optimized the logic.

prathamponnana committed:
Delete my-pic.jpg

prathamponnana committed:
Update app.js

friendly-developer commented on app.js:
```diff
@@ -1,21 +1,25 @@
-    counter2 = localStorage.getItem('counter2');
-}
\ No newline at end of file
+    const storageVal1 = localStorage.getItem('btn1');
```
local storage getter and setter are something that you are using repeatedly all over the file, a better idea would be to move these into a different function, that way you never have to rewrite and test code if we need to move from local storage to session storage.

prathamponnana replied:
Okay sir. Have updated it in the recent comitt.

friendly-developer commented on app.js:
```diff
@@ -1,21 +1,25 @@
-}
\ No newline at end of file
+    const storageVal1 = localStorage.getItem('btn1');
+    const value1 = !storageVal1 && isNan(storageVal1)?0:localStorage.getItem('btn1');
```
these validations can be moved to a util function as well, better you can move all the storage utils to a different file and import that file to use those functions

prathamponnana replied:
Created a new utility function and imported modules from there in the latest comitt. It has increased the readability also now.

friendly-developer commented on styles.css:
```diff
@@ -9,7 +9,7 @@ p{
 }
 
-.my-pic1,.my-pic2{
+.my-pic{
```
good job moving out all of that into a single class

friendly-developer commented on app.js:
```diff
@@ -1,21 +1,25 @@
-    localStorage.setItem('counter2',counter2);
-    document.getElementById('btn2').innerHTML= localStorage.getItem('counter2');
+document.addEventListener('DOMContentLoaded', function(){
+      getCount();
```
A good idea to fetch count on DOM load.
The regular practice is to move onLoad functions to the bottom of the file.

friendly-developer reviewed (changes requested):
Overall, you have done a real good job. 
Although code reviews are never going to be good, there will be always some improvement, I tried to mimic how enterprise reviews works

prathamponnana committed:
Update app.js

Optimized get and set function usage of localStorage.

prathamponnana committed:
Update index.html

prathamponnana committed:
Update app.js

Improved optimization

prathamponnana committed:
Add files via upload

Added a separate utility function and hence improved readability and optimization.

prathamponnana committed:
Update styles.css

prathamponnana commented:
> Overall, you have done a real good job.
> Although code reviews are never going to. be good let's go, there will be always some improvement, I tried to mimic how enterprise reviews works

Thank you sir. Your code reviews are helping me a lot.

friendly-developer commented on app.js:
```diff
@@ -1,25 +1,26 @@
-    const storageVal2 = localStorage.getItem('btn2');
-    const value2 = !storageVal2 && isNan(storageVal2)?0:localStorage.getItem('btn2');
-    document.getElementById('btn2').innerText = value2;
+    let buttons = Array.from(document.getElementsByClassName('btn'));
```
That's a nice way to get all elements, 'btn' is a generic class, try a unique name going forward

friendly-developer commented on app.js:
```diff
@@ -1,25 +1,26 @@
-    currElem.innerText= localStorage.getItem(btn);   
+      const currElem = document.getElementById(btn);
+      const currCount = parseInt(currElem.innerText);
+      localStorage.setItem(btn,currCount+1);
```
you can probably move this as well to the utils, next time

friendly-developer reviewed (approved):
Awesome job. Keep going.👍🏻

friendly-developer commented:
Merge this PR to master and then update the github page in the readme.

prathamponnana commented:
> Merge this PR to master and then update the github page in the readme.

Okay sir. Thank you.
